pkg/spec/deployment_transform.go

```diff
@@ -0,0 +1,120 @@
+/*
```

Comment:
can't we have all the `deployment` related methods in one file ?

Reply:
because those methods are really small and I think having it in one file wil not make that file longer.

Reply:
The methods will grow a lot in subsequent refactors, having them in different files makes it much more readable imo since every method in the interface has its own file.

Reply:
I kind of agree with @surajssd here. I think that having all deployment related functions in one file would be much better. Those files are really small. Even if they grow twice it is still manageable number of lines
```
▶ wc -l deployment_*                                                                            
  61 deployment_fix.go
 120 deployment_transform.go
  33 deployment_unmarshal.go
  31 deployment_validate.go
 245 total
```


I think that we can even put all `resource_*` files into one it is not that much lines:

```
▶ wc -l resources_fix.go resources_validate.go resources_transform.go    
  109 resources_fix.go
   35 resources_validate.go
  275 resources_transform.go
  419 total
```

Or organize them by Kind it will create much more files, but I think it is more logical.

What do you think?

Reply:
ping @containscafeine on this one!

Reply:
@surajssd @kadel ack, I have moved the deployment functions to one file, and resources functions to one file for now. If the resources file gets confusing later on, we can break it down later.